mars-sim-ui/src/main/java/org/mars_sim/msp/ui/swing/tool/monitor/AbstractMonitorModel.java

```diff
@@ -0,0 +1,111 @@
+/*
+ * Mars Simulation Project
+ * AbstractMonitorModel.java
+ * @date 2023-09-18
+ * @author Barry Evans
+ */
+package org.mars_sim.msp.ui.swing.tool.monitor;
+
+import java.io.Serializable;
+
+import javax.swing.table.AbstractTableModel;
+
+import org.mars.sim.tools.Msg;
+
+/**
+ * Default implementation of the MonitorModel
+ */
+public abstract class AbstractMonitorModel extends AbstractTableModel
+        implements MonitorModel {
+
+    /**
+     * Helper class to define the specification of a column
+     */
+    protected record ColumnSpec (String name, Class<?> type) implements Serializable {};
```

Comment:
awesome that you are using `record` now.

Reply:
I have started using them elsewhere as well. An advantage of upgrading the Java version